presto-main/src/main/java/com/facebook/presto/failureDetector/FailureDetector.java

```diff
@@ -13,11 +13,22 @@
  */
 package com.facebook.presto.failureDetector;
 
+import com.facebook.presto.spi.HostAddress;
```

Comment:
FailureDetector now attempts to determine the state of a node based on the last exception encountered.